makefiles/docker.inc.mk

```diff
@@ -89,6 +89,11 @@ DOCKER ?= docker
 DOCKER_VOLUMES_AND_ENV += $(if $(wildcard $(GIT_CACHE_DIR)),-v $(GIT_CACHE_DIR):$(DOCKER_BUILD_ROOT)/gitcache)
 DOCKER_VOLUMES_AND_ENV += $(if $(wildcard $(GIT_CACHE_DIR)),-e GIT_CACHE_DIR=$(DOCKER_BUILD_ROOT)/gitcache)
 
+# Handle worktree by mounting the git common dir in the same location
+_is_git_worktree = $(shell grep '^gitdir: ' $(RIOTBASE)/.git 2>/dev/null)
```

Comment:
No need for guesswork:

```sh
$ git worktree list --porcelain
worktree /home/jcarrano/source/vanillaRIOT
HEAD 77b91b7c9622b679c2212767552709525ea623e9
branch refs/heads/optparse

worktree /home/jcarrano/source/masterRIOT
HEAD 53de520619be195501d58aa90e9313e35b0ba59f
branch refs/heads/at_driver-fix-cast

```

According to the docs "The main worktree is __listed first__, followed by each of the linked worktrees.",